feg/gateway/sbi/sbi.go

```diff
@@ -15,51 +15,91 @@ package sbi
 
 import (
 	"context"
+	"fmt"
 	"net"
 	"net/http"
+	"net/url"
 	"strings"
 	"time"
 
 	"github.com/labstack/echo/v4"
+	"golang.org/x/oauth2"
+	"golang.org/x/oauth2/clientcredentials"
 )
 
-type SbiServer struct {
-	Server     *echo.Echo
-	ListenAddr string
+type NotifierServer struct {
```

Comment:
This can be also used as a N7 server for example in the Mock PCF. It is better to name it differently.

Reply:
My reasoning here is that N7 server is nothing else than a NotifierServer. The difference is that the api already implements the methods.

In fact, I was tempted to use `BaseClientWithNotifier` for the mock PCF server. Same reason as before: I think those http endpoints are nothing else than a client+server under the hood

Reply:
A Notifier Server is essentially a client, that gets notified by a server. But in case of a N7 Server for example Mock PCF, it will be acting as a pure HTTP server and not as a notification endpoint. It would be better to name this struct accordingly where it can be used by both SBI Client and also a SBI server.

Reply:
Ok sounds good

Reply:
REST APIs (also SBI specs) generally makes distinction between REST API requests served and notifications (callbacks) served, though both are HTTP Request handles.  REST API requests handlers are REST Servers and the Notification or Callback handlers are part of REST Client. This piece of code can be used as both REST Server and a REST client (for the notification handler). Naming it as NotifiyServer is confusing (atleast for me). We should probably go a layer below and name it as HTTPServer or HTTPHandler or RESTHandler.